### PR TITLE
scripts, buidl: revisit

### PR DESCRIPTION
**Pull request by tupakkatapa**
Making a couple of necessary changes to the `scripts/build.sh`.

### CODE REVIEW

File: flake.lock
Comment (jhvst) [inline flake.lock:1]: Why is the flake updated here?

File: scripts/buidl.sh [hunk 1]
```diff
@@ -210,6 +210,9 @@ main() {
   # If JSON data is provided, create 'default.nix' from it
   create_default_nix "$json_data" "$default_nix"
 
+  # Stage the changes in 'default.nix'
+  git add "$default_nix"
```
Comment (jhvst) [inline scripts/buidl.sh:214]: Notice that we have to add git as a dependency to the script.

**Review by jhvst — CHANGES_REQUESTED** [review]

File: flake.lock
Comment (tupakkatapa) [reply flake.lock:1]: To apply this hotfix: https://github.com/ponkila/nixobolus/commit/b1fe9b07cdc500a9786092d666715fe9760cb9c5

File: scripts/buidl.sh (hunk 1 above)
Comment (tupakkatapa) [reply scripts/buidl.sh:258]: Added.

File: scripts/buidl.sh [hunk 2]
```diff
@@ -153,19 +159,45 @@ print_output() {
...
+    mkdir -p "$output_path"
+
+    for symlink in "$kexec_tree"/*; do
+      real_path=$(readlink -f "$symlink")
+      new_real_path="$output_path/$(basename "$symlink")"
+      cp -f "$symlink" "$new_real_path"
+    done
+}
+
 create_webui_files() {
```
Comment (jhvst) [inline scripts/buidl.sh:201]: Can we have this as a separate script? We need to run this every time the web server is started.

Comment (tupakkatapa) [reply scripts/buidl.sh:203]: Sure, doing this tomorrow.

**Review by jhvst — APPROVED** [review]
I think this should be merged now and then work on the create_webui_files on a separate PR.

This PR has features that are needed for functioning web UI.

**Pull request merged**